File: backend/app_tuyen_sinh_django/utils/enums/sex.py
from enum import Enum


class Sex(Enum):
    MALE = "Nam"
    FEMALE = "Nữ"
    OTHER = "Khác"

    def __init__(self, name: str) -> None:
        self._value_ = len(__class__)
        self.sex = name
        __class__.range_val = range(0, self._value_)
        return name

    @staticmethod
    def from_id(input_id: int):
        return next(x for x in __class__ if x.value == input_id)

File: app_tuyen_sinh_flutter/lib/utils/enums/sex.dart
import 'package:app_tuyen_sinh_flutter/utils/extensions/enum_extension.dart';
import 'package:app_tuyen_sinh_flutter/utils/mixins/enum_id_mixin.dart';

enum Sex with EnumIdMixin {
  male(0, 'Nam'),
  female(1, 'Nữ'),
  unknown(2, 'Không xác định');

  const Sex(this.id, this.name);

  @override
  final int id;
  @override
  final String name;

  static Sex fromId(id) {
    return Sex.values.fromId(id) as Sex;
  }
}
